src/bin/electrs.rs

```diff
@@ -122,8 +125,29 @@ fn run_server(config: Arc<Config>) -> Result<()> {
             tip = current_tip;
         };
 
+        // FIXME(jamesdorfman): couldn't figure out how to import it from util
+        let log_fn_duration = |fn_name: &str, duration: u128| {
+            let log = json!({
+                "fn_name": fn_name,
+                "duration_micros": duration,
+            });
+            println!("{}", log);
+        };
+
         // Update mempool
-        mempool.write().unwrap().update(&daemon)?;
+
+        let t = Instant::now();
+
+        let old_txids = mempool.read().unwrap().old_txids();
+        let new_txids = daemon
+            .getmempooltxids()
+            .chain_err(|| "failed to update mempool from daemon")?;
+        let old_mempool_txs: HashSet<&Txid> = old_txids.difference(&new_txids).collect();
```

Comment:
I think that naming it `old_mempool_txs` is a bit confusing (unclear that its old txs that no longer exists, seems similar to `old_txids`) and that the previous variable name `to_remove` better conveyed its meaning. It could also be called `mempool_evicted_txids` to make it even clearer.